### PR TITLE
Trev8939/set max extent

### PR DESCRIPTION
**Pull request by TADraeseke**
Create a new sample demonstrating the new maxExtent property

### CODE REVIEW

File: kotlin/set-max-extent/README.metadata.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        "ArcGISMap",
+        "Envelope"
+    ],
+    "language": "kotlin",
+    "relevant_apis": [
+        "ArcGISMap",
+        "Envelope"
+    ],
+    "snippets": [
+        "/src/main/java/com/esri/arcgisruntime/sample/setmaxextent/MainActivity.kt"
```
Comment (shubham7109) [inline kotlin/set-max-extent/README.metadata.json:23]:
```suggestion
        "src/main/java/com/esri/arcgisruntime/sample/setmaxextent/MainActivity.kt"
```

**Review by shubham7109 — CHANGES_REQUESTED** [review]
Just a minor change in the metadata file. 

One note, avoid screenshots with the attribute tab and the "Licensed for developer use", as they are not really helpful info to have in the sample-viewer screenshots. Also, crop the image to 700x400px rather than resize the image. This helps as the image does not feel "stretched"

File: kotlin/set-max-extent/README.metadata.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+    "category": "Maps and scenes",
```
Comment (shubham7109) [inline kotlin/set-max-extent/README.metadata.json:2]:
```suggestion
    "category": "Maps and Scenes",
```
This is causing a crash while running it on the sample-viewer

**Comment by TADraeseke** [comment]
Thanks @shubham7109! I've made those changes to the metadata file and added code to show the map's max extent as a red dotted outline.

I'm not sure we should be removing the attribution bar in the screenshot though, since the data is licensed, we're really not supposed to block or remove it.

**Review by shubham7109 — APPROVED** [review]
Looks good! We can better visualize the max extent behavior with the graphic now! 

Regarding the screenshots, that's a good point and could be worth a discussion with other platforms and maybe standardizing it.

**Comment by shubham7109** [comment]
I took a look at the seekbar approach for the sample and realized that it wouldn't really work well since `newEnvelope.expand` is not a linear change. Instead, it would work better with buttons.
max_extent


I may have gone a little overboard with the design, but it would showcase the `map.maxExtent` property from a sample perspective with straightforward code implementation.

**Review by saratchandrakarumuri — APPROVED** [review]

**Review by shubham7109 — CHANGES_REQUESTED** [review]

**Comment by shubham7109** [comment]
Removed buttons and added a switch for `maxExtent` using the feedback from the review. The readme will probably need to be update to this

File: kotlin/set-max-extent/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+## Use case
+
+When showing map information relevant to only a certain area, you may wish to constrain the user's
+ability to pan or zoom away.
+
+## How to use the sample
+
+The application loads with a map whose maximum extent has been set to the borders of Colorado. Note
+that you won't be able to pan far from the Colorado border or zoom out beyond the minimum scale set
+by the max extent.
```
Comment (shubham7109) [inline kotlin/set-max-extent/README.md:16]:
```suggestion
The application loads with a map whose maximum extent has been set to the borders of Colorado. Note
that you won't be able to pan far from the Colorado border or zoom out beyond the minimum scale set
by the max extent. Use the toggle switch to disable the max extent to freely pan/zoom around the map. 
```

File: kotlin/set-max-extent/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+## How it works
+
+1. Create an `ArcGISMap` object.
+2. Set the maximum extent of the map with `map.maxExtent = envelope`.
+3. Set the map to a `MapView` object.
```
Comment (shubham7109) [inline kotlin/set-max-extent/README.md:22]:
```suggestion
3. Set the map to a `MapView` object.
4. Set `map.maxExtent = null` to disable the maximum extent of the map 
```

Comment (shubham7109) [inline kotlin/set-max-extent/README.md:22]:
```suggestion
3. Set the map to a `MapView` object.
4. Set `map.maxExtent = null` to disable the maximum extent of the map 
```

**Review by shubham7109 — APPROVED** [review]
Few readme change suggestions based on the toggle switch added to the sample.

File: kotlin/set-max-extent/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toTopOf="@+id/constraintLayout"
+        app:layout_constraintHorizontal_bias="0.0"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+    </com.esri.arcgisruntime.mapping.view.MapView>
+
```
Comment (TADraeseke) [inline kotlin/set-max-extent/src/main/res/layout/activity_main.xml:20]:
```suggestion
        app:layout_constraintTop_toTopOf="parent" />

```

File: kotlin/set-max-extent/src/main/java/com/esri/arcgisruntime/sample/setmaxextent/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            renderer = SimpleRenderer(SimpleLineSymbol(SimpleLineSymbol.Style.DASH, Color.RED, 5f))
+        }
+
+        extentSwitch.setOnCheckedChangeListener { _, isChecked ->
+            if (isChecked)
+                // set max extent to the state of Colorado
+                coloradoMap.maxExtent = extentEnvelope
+            else
+                // disable the max extent of the map, map is free to pan around
+                coloradoMap.maxExtent = null
```
Comment (TADraeseke) [inline kotlin/set-max-extent/src/main/java/com/esri/arcgisruntime/sample/setmaxextent/MainActivity.kt:80]: maybe on toggle we should set Text on the TextView to:

"Maximum extent enabled"

"Maximum extent disabled"

File: kotlin/set-max-extent/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<resources>
+    <string name="app_name">Set max extent</string>
+    <string name="factorText">Modify extent by a factor of 0.5</string>
+    <string name="decrease">-</string>
+    <string name="increase">+</string>
+    <string name="updateMap">Update Map</string>
+    <string name="extentText">Enable/Disable the maximum extent</string>
```
Comment (TADraeseke) [inline kotlin/set-max-extent/src/main/res/values/strings.xml:7]: These can be removed

**Review by TADraeseke — COMMENTED** [review]
Thanks for your work on this @shubham7109! Just couple small comments below!

**Comment by TADraeseke** [comment]
LGTM! Merge when ready @shubham7109!

**Pull request merged**